### ISSUE TITLE
ANR in MPUtility.fetchGoogleAdInfo on Android 14 and 15

### ISSUE THREAD

**Issue by fbcouch** [mParticle/mparticle-android-sdk#500]
Hello, since updating our mParticle SDK to 5.56, we are seeing a new ANR pop up, primarily on Google devices running android 14 or higher. Here is the stack trace:

```
main (waiting):tid=1 systid=12039 
       at jdk.internal.misc.Unsafe.park(Native method)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
       at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:447)
       at java.util.concurrent.FutureTask.get(FutureTask.java:190)
       at com.mparticle.internal.MPUtility.fetchGoogleAdInfo(SourceFile:23)
       at com.mparticle.internal.MPUtility.getGoogleAdIdInfo(SourceFile:5)
       at com.mparticle.internal.MPUtility.getAdIdInfo(SourceFile:12)
       at com.mparticle.internal.b.c0(SourceFile:1)
       at com.mparticle.internal.a$b.run(SourceFile:4)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:243)
       at android.os.Looper.loop(Looper.java:338)
       at android.app.ActivityThread.main(ActivityThread.java:8446)
       at java.lang.reflect.Method.invoke(Native method)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
 ```

![Screenshot 2024-08-01 at 10 56 39 AM](https://github.com/user-attachments/assets/f1ab55f8-9c71-42d5-abc9-3fb9b4c2ad72)

**Comment by samdozor** [comment]
Hi @fbcouch - thank you for the report. It's been assigned to the team internally and we'll revert back to you with an ETA or any questions.

**Comment by Mansi-mParticle** [comment]
@fbcouch I have fixed this issue in version 5.56.5. I am closing this issue now. If you still have problems in the latest version, please let us know.
